src/app/src/components/ClaimedFacilitiesDetails.jsx

```diff
@@ -191,27 +189,18 @@ const InputSection = ({
             }
 
             if (isCreatable && isMultiSelect) {
-                const [
-                    optionsInExistingSet,
-                    newlyCreatedOptionsPrime,
-                ] = partition(
-                    value,
-                    v => includes(map(selectOptions, 'value'), v),
-                );
-
-                const mapStringToOption = s => ({ value: s, label: s });
-
-                return concat(
-                    map(optionsInExistingSet, mapStringToOption),
-                    map(newlyCreatedOptionsPrime, mapStringToOption),
-                );
+                return map(value, s => ({ value: s, label: s }));
             }
 
-            window.console.warn('isCreatable && !isMultiSelect case is not yet implemented');
-
-            return [];
+            // isCreatable && !isMultiSelect creates an option object from the value
+            return {
+                value,
+                label: value,
+            };
         })();
 
+        const SelectComponent = isCreatable ? Creatable : Select;
```

Comment:
Since these two components get an identical set of props, we can cut some lines below by conditionally choosing the component here.